crates/api_crud/src/post/create.rs

```diff
@@ -192,3 +195,35 @@ impl PerformCrud for CreatePost {
     .await
   }
 }
+
+/// Check the intersection of user languages and community languages. If it contains only one item,
+/// use that as default language for the post. Otherwise, use "undetermined".
+async fn default_post_language(
+  local_user_id: LocalUserId,
+  community_id: CommunityId,
+  pool: &DbPool,
+) -> Result<LanguageId, LemmyError> {
+  let (user_langs, community_langs, undetermined_lang) = blocking(pool, move |conn| {
+    Ok::<(Vec<LanguageId>, Vec<LanguageId>, LanguageId), LemmyError>((
+      LocalUserLanguage::read(conn, local_user_id)?,
+      CommunityLanguage::read(conn, community_id)?,
+      Language::read_undetermined(conn)?,
```

Comment:
3 queries, when this could be simplified to one using inner join to find the intersection set.

A note, `Language::read_undetermined` could just be a hard-coded LanguageId number rather than a read, since this data comes from a static migration.

Reply:
Fixed, and i was able to remove Language::read_undetermined entirely.